File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.io.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(bf.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int Start = Integer.parseInt(st.nextToken());
        boolean [] visited = new boolean[N+1];
        ArrayList<ArrayList<Integer>> list = new ArrayList<>();
        for(int i = 0; i < N+1; i++){
            list.add(new ArrayList<>());
        }
        for(int i = 0; i < M; i++){
            st = new StringTokenizer(bf.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            list.get(x).add(y);
            list.get(y).add(x);
        }
        for(int i = 0; i<N+1;i++){
            Collections.sort(list.get(i));
        }
        dfs(visited.clone(), list, Start);
        System.out.println();
        bfs(visited.clone(), list, Start);
    }
    static void bfs(boolean[] visited, ArrayList<ArrayList<Integer>> list, int start){
        visited[start] = true;
        Deque<Integer> q = new ArrayDeque<>();
        q.add(start);
        while(!q.isEmpty()){
            int poped = q.removeFirst();
            visited[poped] = true;
            System.out.print(poped+" ");
            for(Integer i: list.get(poped)){
                if(visited[i] != true){
                    q.add(i);
                    visited[i] = true;
                }
            }
        }
    }
    static void dfs(boolean[] visited, ArrayList<ArrayList<Integer>> list, int start){
        System.out.print(start+" ");
        visited[start] = true;
        for(Integer i: list.get(start)){
            if(visited[i] != true){
                dfs(visited,list,i);
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py

from collections import deque
import copy
N, M, V = map(int, input().split())

visited = [False for _ in range(N+1)]
graph = [[] for _ in range(N+1)]

for _ in range(M):
    x, y = map(int, input().split())
    graph[x].append(y)
    graph[y].append(x)

for i in range(1, N+1):
    graph[i].sort()


def dfs(visited, graph, V):
    visited[V] = True
    print(V, end=' ')
    for i in graph[V]:
        if not visited[i]:
            visited[i] = True
            dfs(visited, graph, i)


dfs(copy.deepcopy(visited), graph, V)

print()


def bfs(visited, graph, V):
    q = deque([V])
    visited[V] = True
    while q:
        poped = q.popleft()
        print(poped, end=' ')
        for i in graph[poped]:
            if not visited[i]:
                q.append(i)
                visited[i] = True


bfs(copy.deepcopy(visited), graph, V)